src/Main.elm

```diff
@@ -176,20 +210,32 @@ view model =
         watchDate movie =
             Movie.watchDate movie
                 |> Maybe.withDefault (date 1800 1 1)
+
+        modal =
+            case model.focusedMovie of
+                Movie.Loaded movieDetails ->
+                    movieModal FocusMovie movieDetails
+
+                Movie.Selected movie ->
+                    offlineMovieModal FocusMovie movie
```

Comment:
@MattCheely here and L217 are where the (function? message generator? `FocusMovie` is passed to the modal to allow it to close by calling `FocusMovie NotSelected`